pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -108,7 +108,7 @@ abstract class Analytics {
     final homeDirectory = getHomeDirectory(fs);
     if (homeDirectory == null ||
         !checkDirectoryForWritePermissions(homeDirectory)) {
-      return NoOpAnalytics();
+      throw Exception('Permissions error on the home directory!');
```

Comment:
can you show different error messages depending on if the home dir doesn't exist or if it doesn't have write permissions?